### ISSUE TITLE
FATA[0000] unrecognized image format (SIGSEGV in tonistiigi/fsutil.Walk)

### ISSUE THREAD

**Issue by mathiasbockwoldt** [containerd/nerdctl#1617]
### Description

There are similar issues but none of them solved my specific problem.

I have a very simple `Dockerfile` like the following (the actual content doesn't change the output).

```
FROM node:18-alpine
CMD ["echo", "test"]
```

Having the file in a specific directory and running `nerdctl build -t test . -f Dockerfile` from within that directory will give me

```
> nerdctl build -t test . -f Dockerfile
[+] Building 0.0s (0/2)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xae3952]

goroutine 100 [running]:
github.com/tonistiigi/fsutil.Walk.func1({0xc0005b8600, 0x32}, {0x0, 0x0}, {0xeb92a0, 0xc0005aa030})
        /src/vendor/github.com/tonistiigi/fsutil/walker.go:137 +0xfb2
path/filepath.walk({0xc00020e090, 0x29}, {0xedc700, 0xc0006e89c0}, 0xc000873d48)
        /usr/local/go/src/path/filepath/path.go:450 +0x20d
path/filepath.Walk({0xc00020e090, 0x29}, 0xc000634d48)
        /usr/local/go/src/path/filepath/path.go:517 +0x6c
github.com/tonistiigi/fsutil.Walk({0xed4fb8, 0xc0006dc300}, {0x7ffcdcdd686c, 0x29}, 0xc0006ec140, 0xc0006d4240)
        /src/vendor/github.com/tonistiigi/fsutil/walker.go:109 +0x95d
github.com/tonistiigi/fsutil.(*fs).Walk(0x0, {0xed4fb8, 0xc0006dc300}, 0xc0000181cb)
        /src/vendor/github.com/tonistiigi/fsutil/fs.go:36 +0x3d
github.com/tonistiigi/fsutil.(*sender).walk(0xc0006f2240, {0xed4fb8, 0xc0006dc300})
        /src/vendor/github.com/tonistiigi/fsutil/send.go:147 +0xa6
github.com/tonistiigi/fsutil.(*sender).run.func1()
        /src/vendor/github.com/tonistiigi/fsutil/send.go:60 +0x2e
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /src/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
        /src/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x92
FATA[0000] unrecognized image format
```

Copying the file to any other directory and running the very same command from that other directory results in a working image.

I assume that something (some file or folder) in the specific directory has a "reserved name" that nerdctl tries to open. From the error message it is unclear which file has the unrecognized format.

### Describe the results you received and expected

I expect the above-mentioned command to work in any directory.
Or at least get a message on *what* is wrong (what file, what property that is unexpected), such that I have a realistic chance of fixing the problem.

### What version of nerdctl are you using?

1.0.0

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

Rancher Desktop for Windows

### Host information

Client:
 Namespace:     default
 Debug Mode:    false

Server:
 Server Version: v1.6.8
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Log: fluentd journald json-file syslog
  Storage: native overlayfs
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.102.1-microsoft-standard-WSL2
 Operating System: Rancher Desktop WSL Distribution
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 24.83GiB
 Name: IA-5CD2041382

**Comment by mathiasbockwoldt** [comment]
I meant other issues that contain "unrecognized image format". I don't know if they are related, because the error message is so generic.
https://github.com/containerd/nerdctl/issues?q=is%253Aissue+%22unrecognized+image+format%22

**Comment by mathiasbockwoldt** [comment]
Sorry for these incremental posts!

If I copy the binding.gyp to another directory, it triggers the error if I run `nerdctl build` in that other directory.
If, however, I deleted the binding.gyp in the original directory and create a new file and paste the above-mentioned file content, everything works just fine.

So, my work-around is: Copy the file content, create a new file, paste the file content.

I don't know much about file systems, but could it have anything to do with that? I'm running Windows 10 with NTFS on that particular disk.

For me, personally, the problem was solved by the work-around as stated above. I kept a copy of the "problematic" file in case you have questions.

**Comment by TBBle** [comment]
The stack trace matches https://github.com/tonistiigi/fsutil/pull/120#issuecomment-1067093957 and https://github.com/moby/buildkit/issues/3257. That segfault was probably fixed by https://github.com/tonistiigi/fsutil/pull/122.

This was fixed in [BuildKit v0.10.6](https://github.com/moby/buildkit/releases/tag/v0.10.6), bundled into [nerdctl v1.1.0](https://github.com/containerd/nerdctl/releases/tag/v1.1.0), pulled into [rancher-desktop-wsl-distro v0.32](https://github.com/rancher-sandbox/rancher-desktop-wsl-distro/releases/tag/v0.32), and hence should have been fixed by [Rancher Desktop v1.8.0](https://github.com/rancher-sandbox/rancher-desktop/commit/ea2c5797603c0809a27e565a3ba724121be9e526) (per [release diff](https://github.com/rancher-sandbox/rancher-desktop/compare/v1.7.0...v1.8.0#diff-05d4fb585ec6313e48f291c379c3660ee607d62b4c843230421f5bb619fe7a02)) in [March 2023](https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.8.0).

**Comment by mathiasbockwoldt** [comment]
I didn't encounter this problem ever since the update. Thank you for the follow-up!
